### ISSUE TITLE
"list index out of range" during training

### ISSUE THREAD

**Issue by KIMMINJUNG02** [myshell-ai/MeloTTS#304]
When running the training script for MeloTTs, the process crashes with a error like this.
2025-10-18 08:00:35.004634: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
/usr/lib/python3.12/contextlib.py:105: FutureWarning: `torch.backends.cuda.sdp_kernel()` is deprecated. In the future, this context manager will be removed. Please see `torch.nn.attention.sdpa_kernel()` for the new context manager, with updated signature.
  self.gen = func(*args, **kwds)
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
2025-10-18 08:00:45.621 | INFO     | data_utils:_filter:64 - Init dataset...
100% 438/438 [00:00<00:00, 6089.56it/s]
2025-10-18 08:00:45.694 | INFO     | data_utils:_filter:84 - min: 174; max: 860
2025-10-18 08:00:45.694 | INFO     | data_utils:_filter:85 - skipped: 0, total: 438
buckets: [78, 42, 78, 96, 84, 54, 18]
/usr/local/lib/python3.12/dist-packages/torch/utils/data/dataloader.py:627: UserWarning: This DataLoader will create 16 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
2025-10-18 08:00:45.697 | INFO     | data_utils:_filter:64 - Init dataset...
100% 4/4 [00:00<00:00, 3819.08it/s]
2025-10-18 08:00:45.699 | INFO     | data_utils:_filter:84 - min: 313; max: 626
2025-10-18 08:00:45.699 | INFO     | data_utils:_filter:85 - skipped: 0, total: 4
Using noise scaled MAS for VITS2
Using duration discriminator for VITS2
/usr/local/lib/python3.12/dist-packages/torch/nn/utils/weight_norm.py:144: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)
(torch.Size([10, 192]), torch.Size([8, 192]))
(torch.Size([256, 256]), torch.Size([1, 256]))
list index out of range

I used the config file that was generated after running the preprocessing code and did not modify it.
